src/manifest.rs

```diff
@@ -252,12 +252,34 @@ impl Manifest {
     pub fn insert_into_table(&mut self, table_path: &[String], dep: &Dependency) -> Result<()> {
         let table = self.get_table(table_path)?;
 
-        if table[&dep.name].is_none() {
+        let table_iter = table.clone(); //FIXME: to avoid double borrow issue ... not ideal
```

Comment:
```suggestion
```

Reply:
table.clone was not needed, removed it, and extracted find logic to separate `find_dep` fn.